core/src/mindustry/mod/Mods.java

```diff
@@ -708,6 +711,82 @@ public void contextRun(LoadedMod mod, Runnable run){
         }
     }
 
+    /** Tries to find the config file of a mod/plugin. */
+    @Nullable
+    public ModMeta findMeta(Fi file){
+        Fi metaFile =
+            file.child("mod.json").exists() ?       file.child("mod.json") :
+            file.child("mod.hjson").exists() ?      file.child("mod.hjson") :
+            file.child("plugin.json").exists() ?    file.child("plugin.json") :
+            file.child("plugin.hjson");
+
+        if(!metaFile.exists()){
+            return null;
+        }
+
+        ModMeta meta = json.fromJson(ModMeta.class, Jval.read(metaFile.readString()).toString(Jformat.plain));
+        meta.cleanup();
+        return meta;
+    }
+
+    /** Resolves the loading order of a list mods/plugins using their internal names.
+     * It also skips non-mods files or folders. */
+    public Seq<Fi> resolveDependencies(Seq<Fi> files){
+        ObjectMap<String, Fi> fileMapping = new ObjectMap<>(files.size);
+        ObjectMap<String, Seq<String>> dependencies = new ObjectMap<>(files.size);
+
+        for(Fi file : files){
+            Fi zip = file.isDirectory() ? file : new ZipFi(file);
+
+            if(zip.list().length == 1 && zip.list()[0].isDirectory()){
+                zip = zip.list()[0];
+            }
+
+            ModMeta meta = findMeta(zip);
```

Comment:
findMeta can throw an exception due to malformed JSON. You should try/catch here and continue if it throws.